### PR TITLE
add popup explaining how to analyze branches

### PR DESCRIPTION
**Pull request by stas-vilchik**
@gregaubert please review only the last commit

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [branch](https://github.com/SonarSource/sonarqube/blob/c3731e978de76154e02c7e67e3357517c7fe8da4/server/sonar-server/src/main/java/org/sonar/server/branch): Add a 'package-info.java' file to document the 'branch' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx [hunk 1]
```diff
@@ -60,38 +64,84 @@ export default class ComponentNavBranch extends React.PureComponent<Props, State
...
   closeDropdown = () => {
     if (this.mounted) {
-      this.setState({ open: false });
+      this.setState({ dropdownOpen: false });
     }
   };
 
-  render() {
+  togglePopup = (show?: boolean) => {
+    if (this.mounted) {
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx]: Is it really useful to check to `mounted` here ? This method is not used asynchronously right ?

File: server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx [hunk 2]
```diff
@@ -60,38 +64,84 @@ export default class ComponentNavBranch extends React.PureComponent<Props, State
...
+  };
+
+  handleSingleBranchClick = (event: React.SyntheticEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.togglePopup();
+  };
+
+  renderDropdown = () => {
+    return this.state.dropdownOpen
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx]: You can drop the `return`

Comment (stas-vilchik) [inline server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx]: I prefer to keep it for consistency with next method.

File: server/sonar-web/src/main/js/store/appState/duck.ts [hunk 3]
```diff
@@ -17,62 +17,52 @@
...
-/*::
-type AppState = {
-  adminPages?: Array<*>,
-  authenticationError: boolean,
-  authorizationError: boolean,
-  organizationsEnabled: boolean,
-  qualifiers: ?Array<string>
-};
-*/
+interface AppState {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/store/appState/duck.ts]: I haven't changed this file in the end, but I decided to not throw my work away 😄

**Comment by stas-vilchik** [comment]
@gregaubert I've added one more popup when there is no plugin to analyze branches

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [branch](https://github.com/SonarSource/sonarqube/blob/76308c2a18510a26c3b44c1edde6e8e42019e3ce/server/sonar-server/src/main/java/org/sonar/server/branch): Add a 'package-info.java' file to document the 'branch' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/projectbranch/ws/BranchDtoToWsBranch.java [hunk 4]
```diff
@@ -45,14 +44,13 @@ private BranchDtoToWsBranch() {
   static WsBranches.Branch.Builder toBranchBuilder(BranchDto branch, Optional<BranchDto> mergeBranch, Map<String, MeasureDto> measuresByMetricKey) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectbranch/ws/BranchDtoToWsBranch.java]: MINOR Specify a "BranchDto" parameter instead. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**